moving_zeroes/moving_zeroes.py

```diff
@@ -3,13 +3,36 @@
 Returns: a List of integers
 '''
 def moving_zeroes(arr):
-    # Your code here
+    # Iterate on the enumerated array
+    # if the value is zero, remove it from the array and append it
+    # check to see if all elements [index:] are zeroes, if so stop.
 
-    pass
+    # Upon testing, it was necessary to wait to pop the zeroes until the
+    # end of the array was reached. To do this:
+    # keep track of the indexes of the zeroes, then when the
+    # end is reached, go back and pop each zero from the list.
+    # The indexes change as the popping occurs -- account for that.
 
+    # list to keep track of zero indexes in the array
+    zero_indexes = []
+
+    for index, element in enumerate(arr):
+        # if the rest of the list is all zeroes, then clean up the list
+        # by popping the zeroes we have come across and return the result
+        if all(zero_check == 0 for zero_check in arr[index:]):
+            for count, zero_index in enumerate(zero_indexes):
+                arr.pop(zero_index - count)
+            return arr
+        # else, if the current element is zero, store its index
+        # and append a zero to the current array
+        elif element == 0:
+            zero_indexes.append(index)
+            arr.append(element)
+    return arr
```

Comment:
Okay, this is a really clever first-pass solution! Since we just had a lecture on optimization, there is a way to achieve this with `O(1)` space-complexity.

If you need a hint, let me know and I can give you one. But, I will say that this is one of those optimizations where you think about your approach vs. re-working what you have.

But, really nice job with this!